### PR TITLE
Add puppi multiplicities to MiniAOD

### PR DESCRIPTION
**Pull request by ahinzmann**
PUPPI-weighted multiplicities were found useful to define pileup-safe jet ID criteria and will shortly be recommended for analyses as discussed during this presentation:
https://indico.cern.ch/event/707753/contributions/2905460/attachments/1607520/2551266/JetID_run2017_AK4_PUPPI_recommendations_27022018_v1.pdf
We would therefore like to include them into MiniAOD (and backport to 94 asap).

It was checked that the new userfloats are added correctly to the slimmedJetsPuppi collection running the standard MiniAOD sequence.

Timing and size was checked using test/patMiniAOD_standard_cfg.py with 1000 events of RelValZEE_13/GEN-SIM-RECO/PU25ns_92X_upgrade2017_realistic_v7-v1:

Without this PR:
TimeReport   0.000082     0.000082     0.000082  slimmedJetsPuppi
patJets_slimmedJetsPuppi__PAT. 5769.94 668.333

With this PR:
TimeReport   0.000235     0.000235     0.000235  slimmedJetsPuppi
TimeReport   0.000071     0.000071     0.000071  slimmedJetsPuppiNoMultiplicties
TimeReport   0.000054     0.000054     0.000054  patPuppiJetSpecificProducer
patJets_slimmedJetsPuppi__PAT. 6953.5 705.626

### CODE REVIEW

**Comment by slava77** [comment]
@ahinzmann 
please provide estimates of the size on disk needed for this PR per event. 
CPU per event would be nice to know as well.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by ahinzmann** [comment]
@slava77 I've added the timing and size to the description of the PR

**Comment by perrotta** [comment]
Crash looks like being correlated:

```
----- Begin Fatal Exception 28-Feb-2018 22:31:01 CET-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
   [1] Running path 'FEVTDEBUGHLToutput_step'
   [2] Prefetching for module PoolOutputModule/'FEVTDEBUGHLToutput'
   [3] Calling method for module PATJetSelector/'selectedUpdatedPatJetsPuppiJetSpecific'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for a container with elements of type: pat::Jet
Looking for module label: updatedPatJetsPuppiJetSpecific
Looking for productInstanceName: 

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
```

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
please test

The failing AddOn tests without any error message reported besides
```
A fatal system signal has occurred: external termination request
The following is the call stack containing the origin of the signal.
```
(and nothing else that follows) looks like the glitch that is showing up quite often in the last days: let try to re-run the tests

File: PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py [hunk 1]
```diff
@@ -423,10 +423,26 @@ def miniAOD_customizeCommon(process):
 
     task.add(process.slimmedJets)
     task.add(process.slimmedJetsAK8)
-    addToProcessAndTask('slimmedJetsPuppi', process.slimmedJetsNoDeepFlavour.clone(), process, task)
-    process.slimmedJetsPuppi.src = cms.InputTag("selectedPatJetsPuppi")    
-    process.slimmedJetsPuppi.packedPFCandidates = cms.InputTag("packedPFCandidates")
 
+    addToProcessAndTask('slimmedJetsPuppiNoMultiplicties', process.slimmedJetsNoDeepFlavour.clone(), process, task)
```
Comment (perrotta) [inline PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py]: Is the mispelling ("Multiplicties") wanted? If not, please fix

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- Six userfloats added per jet to slimmedJetsPuppi
- Differences reported in the jenkins tests refer to the newly added userfloats to the miniAOD output
- miniAOD event size on disk increases by ~40 B (on slimmedJetsPuppi) when measured on Zee events with PU (as reported in the PR description)

**Comment by fabiocos** [comment]
+1

**Comment by perrotta** [comment]
@fabiocos : this PR still misses the "analysis" signature and you need a explicit "merge" if you want to merge it

**Comment by fabiocos** [comment]
merge

**Comment by arizzi** [comment]
for the records: me and  @gpetruc understood that we have to sign for AnalysisTools related PRs in PhysicsTools/PatAlgos or NanoAODTools only in the MAOD and analysis branches, not on master. Is that right?

**Comment by fabiocos** [comment]
yes, I would like your explicit approval for the AN branch, that is the general request. As far as PhysicsTools/NanoAOD is concerned anyway, I would say that missing an independent analysis signature it makes sense that you sign everywhere, as you are anyway the responsible and the tester of it

**Pull request merged**